### ISSUE TITLE
CBC return status not handled correctly

### ISSUE THREAD

**Issue by jackvreeken** [casadi/casadi#2410]
Currently the CbcInterface says
```cpp
    m->success = m->return_status==0 && model.isProvenOptimal();
```
According to some examples in Cbc, the correct procedure is "*[...] but initially check if status is 0 and secondary status is 1 -> infeasible*". So it should instead be more like
```cpp
    m->success = m->return_status==0 && !model.secondaryStatus();
```
However, note the "but" in the explanation of CBC. I have no idea why they make it so complicated, but I run into some cases where CBC would return infeasible according to that criterion, but isn't actually infeasible.

Example of some return statuses when problem is infeasible:
```
status: 0, isProvenOptimal: 0, secondaryStatus: 1
isProvenInfeasible: 1, objective: <some ridiculously large value>
```

Makes sense I guess.

Now I have MILP problem, and I have `hot_start` enabled (meaning `setBestSolution` is called with with the x0). My seed is so good that `setBestSolution` with `check=True` finds that the objective for that guess is 0.0. Then it goes on to actually solve by calling `CbcMain1`, which is not able to find a better solution, resulting in the secondary return status being 1 ("*linear relaxation not feasible (or worse than cutoff)*"). According to the simple criterion above this would be considered infeasible, but now look at all return values:
```
status: 0, isProvenOptimal: 1, secondaryStatus: 1
isProvenInfeasible: 0, objective: 0.0000000
```
Note that isProvenOptimal is 1, isProvenInfeasible is 0, and the objective has a proper value. Alright, so lets make the check a bit more complicated, by including all three values on the first line. But of course it would not be that easy... It turns out the solution returned by `model.getColSolution()` after the `CbcMain1` call is worthless in this case, and that's the one that CasADi returns to the user.

I'm not sure how to resolve this, but a couple things we should probably figure out/decide on are:
1. What is the proper combination to check if the solve was succesful. We probably need to check at least 3 values: status, secondaryStatus, and isProvenOptimal.
2. Is there some option for Cbc to just return a known correct/optimal solution it gets this infeasible/cutoff error internally.
3. If not, and the user is using hotstart, can and should we get and store the best solution ourselves (if it is actually feasible), and return that in the case outlined above (status=0, secondary=1, optimal=1)? Maybe this combination does not always mean that the optimized guess is optimal, and maybe there are other combinations that also mean the same thing.

**Comment by jaeandersson** [comment]
I'm not sure how much the Cbc interface is being used in general and how much thought has gone in to that particular line of code. If you can motivate a better solution, go ahead and make a pull request.

**Comment by jgillis** [comment]
Was fixed by @jackvreeken  in #2448
